### PR TITLE
Add GPU worker support

### PR DESCRIPTION
**Pull request by fabeschan**
- [x]  detect number of gpus on worker

- [x] add gpu info to checkin request

- [x] migrate database (add gpu column to worker table)

- [x] add functionality to filter workers based on requested_gpus

### CODE REVIEW

File: worker/worker.py [hunk 1]
```diff
@@ -104,13 +107,24 @@ def _get_memory_bytes(self):
             # Fallback to sysctl when os.sysconf('SC_PHYS_PAGES') fails on OS X
             return int(check_output(['sysctl', '-n', 'hw.memsize']).strip())
 
+    def _get_gpu_count(self):
+        if not self._docker._use_nvidia_docker:
+            return 0
+
+        container_id = self._docker.run_nvidia_smi('-L', 'gcr.io/tensorflow/tensorflow:latest-gpu')
```
Comment (kashizui) [inline worker/worker.py]: I think we can use a more basic image than the tensorflow one? Maybe one of these:
https://hub.docker.com/r/nvidia/cuda/

File: worker/docker_client.py [hunk 2]
```diff
@@ -229,6 +229,35 @@ def download_image(self, docker_image, loop_callback):
                 loop_callback(status)
 
     @wrap_exception('Unable to start Docker container')
+    def run_nvidia_smi(self, args, docker_image):
```
Comment (kashizui) [inline worker/docker_client.py]: I think I can see why but wondering what your reasoning was for running 'nvidia-smi' inside Docker rather than directly?

**Review by kashizui — CHANGES_REQUESTED** [review]
Overall looks pretty good, and it works!

File: worker/docker_client.py (hunk 2 above)
Comment (fabeschan) [reply worker/docker_client.py]: I think in the future we're going to deploy workers as docker containers, in which case running `nvidia-smi` inside docker would not require us to have two different versions of worker images (one for gpu, one not) to deploy.

File: worker/worker.py (hunk 1 above)
Comment (fabeschan) [reply worker/worker.py]: That's a great suggestion.

File: worker/docker_client.py (hunk 2 above)
Comment (fabeschan) [reply worker/docker_client.py]: Done!

Comment (kashizui) [reply worker/docker_client.py]: Ah that's very inception-y to think about haha.

**Review by kashizui — APPROVED** [review]

File: worker/docker_client.py (hunk 2 above)
Comment (fabeschan) [reply worker/docker_client.py]: ;)

**Pull request merged**